Makefile

```diff
@@ -10,8 +10,11 @@ PROC?=$(shell uname -m)
 
 CC=gcc 
 CFLAGS=-fPIC -O3 -fomit-frame-pointer -fno-exceptions -Wall -std=c99 -pedantic
+MODEL=$(cat /proc/device-tree/model)
 ifeq (${PROC},x86_64)
 	CFLAGS+=-m64 -mtune=generic
+else ifneq (,$(findstring "Raspberry Pi", $(MODEL)))
```

Comment:
I would check the PROC var, or at least leverage on uname:

>pi@raspberrypi:~ $ uname -a
>Linux raspberrypi 4.1.19-v7+ #858 SMP Tue Mar 15 15:56:00 GMT 2016 armv7l GNU/Linux

I don't have a rpi now to check what `uname -m` outputs (basically the value of $PROC), you may want to look at it.

Right now (without your mods), what happens?

Reply:
`gcc: error: unrecognized argument in option ‘-march=i686’`

Reply:
ok , `uname -m` return `armv7l`

Reply:
ok, so please update the PR (and rebase it to have a single commit) and I'll be happy to check it again and merge!

Reply:
This is pretty much a duplicate of my pr #19 
Adding that specific raspberry pi code is not needed afaik.

Just need to eliminate the usage of m32 etc from non x86 based architectures like arm etc.

Reply:
@leonmak I just merged @MartB #19 , could you give it a shot and check if this can be closed?